### PR TITLE
Fix creating OperatingSystem with PlatformID.Other

### PR DESCRIPTION
**Pull request by akoeplinger**
Creating an OperatingSystem with PlatformID.Other would result in an exception about an unexpected enum value.
https://github.com/dotnet/runtime/pull/38790 was missing some tests that verified this behavior.

### CODE REVIEW

**Review by stephentoub — COMMENTED** [review]
Is there a test validating that other is returned on browser?

**Comment by akoeplinger** [comment]
@stephentoub there was one in System.Runtime.Extensions and it was actually failing but we're not running it on CI until https://github.com/dotnet/runtime/pull/38996 is merged. I fixed it to verify the expected value and tested locally.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs [hunk 1]
```diff
@@ -65,6 +65,7 @@ public string VersionString
                         case PlatformID.Unix: os = "Unix "; break;
                         case PlatformID.Xbox: os = "Xbox "; break;
                         case PlatformID.MacOSX: os = "Mac OS X "; break;
+                        case PlatformID.Other: os = "Other"; break;
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs:68]: Does this need a space after it? The others have a space before the version.

Comment (akoeplinger) [reply src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs:68]: Thanks for spotting this, opened https://github.com/dotnet/runtime/pull/39180.

**Pull request merged**